### PR TITLE
Additional cleanup

### PR DESCRIPTION
**Pull request by quasiben**
PR cleans up a few remaining items before release:
- Metdata in pyproject.toml
- Remove skips and update tests

### CODE REVIEW

**Comment by brandon-b-miller** [comment]
Does this close https://github.com/rapidsai/pynvjitlink/issues/4 ?

**Review by brandon-b-miller — APPROVED** [review]

File: pynvjitlink/tests/test_pynvjitlink.py [hunk 1]
```diff
@@ -49,30 +49,26 @@ def undefined_extern_cubin():
     return read_test_file('undefined_extern.cubin')
 
 
-@pytest.mark.skip
```
Comment (wence-) [inline pynvjitlink/tests/test_pynvjitlink.py:52]: I think this was fixed in CTK 12.2, but we're not installing environments with that yet, so I am not sure if we can skip these?

Comment (brandon-b-miller) [reply pynvjitlink/tests/test_pynvjitlink.py:52]: We require cuda 12.3 to build the project now https://github.com/rapidsai/pynvjitlink/blob/main/CMakeLists.txt#L14

Comment (wence-) [reply pynvjitlink/tests/test_pynvjitlink.py:52]: Ah, OK. Is that also the version that the end user gets?

Comment (brandon-b-miller) [reply pynvjitlink/tests/test_pynvjitlink.py:52]: Yes - we build in the two pieces of the CTK that are used statically so the user should end up with the 12.3 ctk binaries on their system regardless of what they have installed.

**Review by gmarkall — APPROVED** [review]
Changes look good to me.

**Review by wence- — APPROVED** [review]

**Pull request merged**